airflow/www/forms.py

```diff
@@ -230,14 +236,14 @@ class ConnectionForm(DynamicForm):
     port = IntegerField(lazy_gettext('Port'), validators=[Optional()], widget=BS3TextFieldWidget())
     extra = TextAreaField(lazy_gettext('Extra'), widget=BS3TextAreaFieldWidget())
 
-    # You should add connection form fields in the hooks via get_connection_form_widgets() method
-    # See for example airflow/providers/jdbc/hooks/jdbc.py
-    # It is esed to customized the form, the forms elements get rendered
-    # and results are stored in the extra field as json. All of these
-    # need to be prefixed with extra__ and then the conn_type ___ as in
-    # extra__{conn_type}__name. You can also hide form elements and rename
-    # others from the connection_form.js file
+    # Used to store a dictionary of field customizations used to dynamically change available
+    # fields in ConnectionForm based on type of connection chosen
+    field_parameters = HiddenField(
+        lazy_gettext('field_parameters'), default=json.dumps(_get_field_customizations())
+    )
```

Comment:
Making this a field means this gets submitted back to the server which isn't needed.

Please find another way to get this in to the front end.

Reply:
I think you can do this with a custom widget:

```python
class provider_field_widget(field, **kwargs):
        return Markup('<script id="{}" type="text/json">{}</script>').format(
            field.name,
            json.dumps(ProvidersManager().field_customizations)
        )
```

And then:

```suggestion

    field_parameters = HiddenField(
        lazy_gettext('field_parameters'), widget=provider_field_widget
    )
```

(And `$("#field_parameters").text()` would continue to work)

Reply:
Yep. Good one. Will try it :)

Reply:
I could not make it works (rendering scripts in submition forms basically does not work), but I found a much better way. Instead of modifying the Model, I modified the View and FormWidget. I learned a bit how FAB works internally and it turned out that form widget is a better (and more natural) way to pass the field behaviour.

That actually led me to a code where ConnectionForm does not have any customizations, and all is done in the view part And it also let me to better place where to initialze the providers and I am doing it now after plugin initialization.

I decided to merge those two PRs, they make much more sense together than separately.